### PR TITLE
Redis

### PR DESCRIPTION
**Pull request by renanaAkeyless**

### CODE REVIEW

File: charts/akeyless-api-gateway/templates/cache.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        env:
+        - name: REDIS_DISABLE_COMMANDS
+          value: "FLUSHDB,FLUSHALL"
+        - name: REDIS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "akeyless-api-gw.cacheSecretName" . }}
+              key: cache-pass
+{{- end}}
+
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/cache.yaml:84]: remove spaces

File: charts/akeyless-api-gateway/templates/cache.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+---
+{{- if eq (include "akeyless-api-gw.adminAccessUidExist" .) "true" }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/cache.yaml:87]: this if is unnecessary, the first if wraps both the deployment and service

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 3]
```diff
@@ -144,6 +151,29 @@ spec:
             - name: CONFIG_PROTECTION_KEY_NAME
               value: {{ .Values.akeylessUserAuth.configProtectionKeyName }}
           {{- end }}
+          {{- if eq (include "akeyless-api-gw.adminAccessUidExist" .) "true"  }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:154]: Put both of the env vars ADMIN_UID_TOKEN and GATEWAY_CLUSTER_CACHE in one if statement of {{- if eq (include "akeyless-api-gw.adminAccessUidExist" .) "true" }}

line 114

File: charts/akeyless-api-gateway/templates/cache.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        ports:
+        - containerPort: 6379
+          name: cache
+        livenessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - |            
+                #!/usr/bin/env bash -e
```
Comment (liorhasson) [inline charts/akeyless-api-gateway/templates/cache.yaml:38]: from where did you took this base script? can you add a link reference

File: charts/akeyless-api-gateway/templates/cache.yaml (hunk 2 above)
Comment (renanaAkeyless) [reply charts/akeyless-api-gateway/templates/cache.yaml:87]: done

File: charts/akeyless-api-gateway/templates/cache.yaml (hunk 1 above)
Comment (renanaAkeyless) [reply charts/akeyless-api-gateway/templates/cache.yaml:84]: done

File: charts/akeyless-api-gateway/templates/cache.yaml (hunk 4 above)
Comment (renanaAkeyless) [reply charts/akeyless-api-gateway/templates/cache.yaml:38]: https://stackoverflow.com/questions/68304279/incorrect-liveness-probe-for-redis-not-failing

File: charts/akeyless-api-gateway/templates/deployment.yaml [hunk 5]
```diff
@@ -148,6 +148,34 @@ spec:
...
+              value: enable
+            - name: REDIS_ADDR
+              value: {{  include "akeyless-api-gw.fullname" . }}-cache-svc:6379
+            - name: REDIS_PASS
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "akeyless-api-gw.cacheSecretName" . }}
+                  key: cache-pass
+          {{- end }}
+          {{- if .Values.universalIdentity.uidRotationInterval }}
```
Comment (niv-ez) [inline charts/akeyless-api-gateway/templates/deployment.yaml:167]: If universal identitiy is depandent in "caching.enabled" then we should add a "required" statement for that. Also mention it in comment in the values.yaml file and readme

Comment (renanaAkeyless) [reply charts/akeyless-api-gateway/templates/deployment.yaml:167]: caching.enabled ?  there is no filed named caching.enabled...
 redis is internal caching and the customer doesn't have any interaction with it or any say about it, 
there is no filed in values.yaml to control the redis,

Comment (niv-ez) [reply charts/akeyless-api-gateway/templates/deployment.yaml:167]: Check api-gw values.yaml file

**Review by niv-ez — APPROVED** [review]

**Review by liorhasson — APPROVED** [review]

**Pull request merged**